### PR TITLE
Pubkey and ABI handlers

### PR DESCRIPTION
**Pull request by pikonha**
# Title

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Changes
- [ ] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [ ] Other (please specify)

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: packages/gateway/src/handlers/pubkey.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  setPubkey(params: SetPubkeyProps)
+}
+
+export function withSetPubkey(
+  repo: WriteRepository,
+  validator: OwnershipValidator,
+): ccip.HandlerDescription {
+  return {
+    type: 'setPubkey(bytes32 node,bytes32 x, bytes32 y)',
+    func: async ({ node, x, y }, { signature }) => {
```
[the remark this answers is not in the document]
Comment (pikonha) [reply packages/gateway/src/handlers/pubkey.ts:21]: public key coordinates on the elliptic curve

File: packages/gateway/src/handlers/pubkey.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      try {
+        const isOwner = await validator.verifyOwnership({
+          node,
+          signature: signature! as TypedSignature,
+        })
+        if (!isOwner) {
+          return { error: { message: 'Unauthorized', status: 401 } }
+        }
+
+        await repo.setPubkey({ node, x, y })
```
[the remark this answers is not in the document]
Comment (pikonha) [reply packages/gateway/src/handlers/pubkey.ts:31]: public key coordinates on the elliptic curve

File: packages/gateway/src/handlers/pubkey.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  getPubkey(params: GetPubkeyProps): Promise<GetPubkeyResponse | undefined>
+}
+
+export function withGetPubkey(repo: ReadRepository): ccip.HandlerDescription {
+  return {
+    type: 'pubkey(bytes32 node)',
+    func: async ({ node }) => {
+      const pubkey = await repo.getPubkey({ node })
+      if (pubkey) {
+        const { x, y } = pubkey.value
```
[the remark this answers is not in the document]
Comment (pikonha) [reply packages/gateway/src/handlers/pubkey.ts:49]: public key coordinates on the elliptic curve

File: packages/contracts/src/DatabaseResolver.sol [hunk 4]
```diff
@@ -136,13 +136,13 @@ contract DatabaseResolver is
...
      * Resolves a name, as specified by ENSIP 10 (wildcard).
-     * @param name The DNS-encoded name to resolve.
+     * @param - name The DNS-encoded name to resolve.
      * @param data The ABI encoded data for the underlying resolution function
      * (Eg, addr(bytes32), text(bytes32,string), etc).
      * @return The return data, ABI encoded identically to the underlying function.
      */
     function resolve(
-        bytes calldata name,
+        bytes calldata, /* name */
```
[the remark this answers is not in the document]
Comment (LeonardoVieira1630) [reply packages/contracts/src/DatabaseResolver.sol:145]: I like it. It is a good pattern even though i have never seen some one doing like this.

**Pull request merged**